source/include/ble/cs_BleCentral.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Mar 26, 2021
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#pragma once
+
+#include <ble/cs_Nordic.h>
+#include <ble/cs_UUID.h>
+#include <common/cs_Types.h>
+#include <events/cs_Event.h>
+#include <structs/cs_PacketsInternal.h>
+#include <events/cs_EventListener.h>
+
+/**
+ * Class that enables you to connect to a device, and perform write or read operations.
+ *
+ * Generally, you first connect, then discover, then perform read and/or write operations, and finally disconnect.
+ * Note that the disconnect event might happen at any time.
+ */
+class BleCentral: EventListener {
+public:
+	static BleCentral& getInstance() {
+		static BleCentral instance;
+		return instance;
+	}
+
+	void init();
+
+	/**
+	 * Connect to a device.
+	 *
+	 * @return ERR_BUSY                When already connected or another operation is in progress (discovery, read, write, connect, disconnect).
```

Comment:
can we split out 'allready connected' as a status?

Reply:
Ah, yes, a different error code might be handy.

Reply:
Turns out it was already the case :D